pkg/cloudinit/common.go

```diff
@@ -190,3 +201,32 @@ func getSnapstoreProxyConfigFiles(data BaseUserData) []File {
 
 	return []File{schemeFile, domainFile, storeIDFile}
 }
+
+// getProxyConfigFiles returns the node proxy config files.
+// If the HTTPProxy or HTTPPSProxy is not set, it returns nil.
+// Nil indicates that no files are returned.
+func getProxyConfigFiles(data BaseUserData) []File {
+	if data.HTTPProxy == "" && data.HTTPSProxy == "" {
+		return nil
+	}
+	return []File{
```

Comment:
Let's check if the variable is provided to add the respective file to the list. For example
```
if data.HTTPProxy != "" {
   files = append(files, File{...})
}
```